packages/types/src/transfer.ts

```diff
@@ -5,6 +5,12 @@ export interface RequestHandler<
   ResponseType,
   HandlerOptions = {}
 > {
+  /**
+   * metadata contains information of a handler. For example
+   * 'h2' refers this handler is for handling HTTP/2 requests,
+   * whereas 'h1' refers handling HTTP1 requests
+   */
+  metadata?: Array<string>;
```

Comment:
Change from Array to an object, create a new interface say `RequestHandlerMetaData`

```typescript
export interface RequestHandlerMetadata {
  handlerProtocol: string;
}
```